packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -314,39 +326,75 @@ export default class ConfigParser {
     }
 
     /**
-     * returns a flatten list of globed files
+     * returns a flattened list of globbed files
      *
-     * @param  {String[]} filenames  list of files to glob
-     * @return {String[]} list of files
+     * @param  {String[] | String[][]} filenames list of files to glob
+     * @param  {Boolean} flag to indicate omission of warnings
+     * @param  {FileSystemPathService} file system path service for expanding globbed file names
+     * @param  {number} hierarchy depth to prevent recursive calling beyond a depth of 1
+     * @return {String[] | String[][]} list of files
      */
-    static getFilePaths (patterns: string[], omitWarnings?: boolean, findAndGlob: CurrentPathFinder & Globber & DeterminesAbsolutePath = new FileSystemPathService()) {
-        let files: string[] = []
+    static getFilePaths(patterns: (string | string[])[], omitWarnings?: boolean, findAndGlob: CurrentPathFinder & Globber & DeterminesAbsolutePath = new FileSystemPathService(), hierarchyDepth?: number) {
+        let files: (string | string[])[] = []
+        let groupedFiles: string[] = []
+        let depth: number = hierarchyDepth || 0
 
         if (typeof patterns === 'string') {
             patterns = [patterns]
         }
 
+        // Check we have an array
         if (!Array.isArray(patterns)) {
-            throw new Error('specs or exclude property should be an array of strings')
+            throw new Error('specs or exclude property should be an array of strings, specs may also be an array of string arrays')
         }
 
-        patterns = patterns.map(pattern => removeLineNumbers(pattern))
+        patterns = patterns.map(pattern => {
+            if (Array.isArray(pattern)) {
+                return pattern.map(subPattern => removeLineNumbers(subPattern))
+            }
+            return removeLineNumbers(pattern)
+        })
 
         for (let pattern of patterns) {
-            let filenames = findAndGlob.glob(pattern)
-            filenames = filenames.filter(
-                (filename) => SUPPORTED_FILE_EXTENSIONS.find(
-                    (ext) => filename.endsWith(ext)))
+            // If pattern is an array, then call getFilePaths again
+            // But only call one level deep, can't have multiple levels of hierarchy
+            if (Array.isArray(pattern) && depth === 0) {
```

Comment:
Can't we just do 
```suggestion
            if (Array.isArray(pattern) && hierarchyDepth) {
```
and get rid of `depth`?

Reply:
Agreed